### PR TITLE
documented all the things

### LINKED ISSUE #41: Write up something explaining how others can create plugins?
Hey @callmehiphop 

Can we put together a readme of sorts that explains how someone new would get bootstrapped and able to start building plugins?

@bokowski wants to poke around and needs a bit of guidance.

### PR DESCRIPTION
**Pull request by callmehiphop**
closes #41 

This update shows how to run a local instance of Phoenix and gives examples on how to develop plugins for it. I'm opening this as PR just because I'd love any suggestions on additional info  you think the README could use :)

[preview](https://github.com/callmehiphop/phoenix/blob/docs/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,205 @@
...
+
+### Create repo for your plugin
+
+**NOTE:** Currently we only support plugins hosted on public GitHub
+repositories.
+
+Your plugin name should be very specific to what it does, since we're using
+Angular 1.x modules for plugins we need to avoid naming collisions. For example,
+if your plugin is named `storage-browser` you should prefix any
+controller/view/services with that name. (`StorageBrowserCtrl`, etc.)
```
Comment (stephenplusplus) [inline README.md]: Can we suggest a "." or "-" or something as a more namespace-ish separator, or does that not work in Angular land?

File: README.md [hunk 2]
```diff
@@ -1 +1,205 @@
...
+```javascript
+{
+  "files": [
+    "dist/storage-browser.js",
+    "dist/storage-browser.css"
+  ]
+}
+```
+
+Currently we support, html, css and js. We recommend concatenating your files
```
Comment (stephenplusplus) [inline README.md]:
```diff
- Currently we support,
+ Currently we support
```

File: README.md [hunk 3]
```diff
@@ -1 +1,205 @@
...
+
+### Create your module
+
+Phoenix uses [System.js](https://github.com/systemjs/systemjs) under the hood to
+import your plugin files, so you'll want to export your angular module.
+
+```javascript
+var mod = angular.module('my-plugin', []);
+
+if (typeof module !== 'undefined' && module.exports) {
```
Comment (stephenplusplus) [inline README.md]: Honest curiosity: why this stuff?

**Comment by stephenplusplus** [comment]
This is inspiring! We do a lot of cool stuff with this project. I think as we go, we can present it a little better and the generator will help. We should probably have a simple docs site in the future just to go over this stuff, too.

Anyway, looks good to me! @jgeewax?

File: README.md (hunk 1 above)
Comment (callmehiphop) [inline README.md]: sure can!

File: README.md (hunk 3 above)
Comment (callmehiphop) [inline README.md]: System.js is an AMD loader, so you need to export something in order it for it to pass back a reference.

Comment (stephenplusplus) [inline README.md]: Alrighty. Just a bummer this specific like always needs to be present.

Comment (callmehiphop) [inline README.md]: Well, I'm not married to System.js or anything, so if we want to try and cook up something that removes this, that's fine by me :)

**Comment by callmehiphop** [comment]
Merging because it's been a hot minute and I haven't really heard anything in regards to this.

**Pull request merged**